slosh/run-tests.slosh

```diff
@@ -1,26 +1,34 @@
 #!/usr/bin/env slosh
 
+(load "init.slosh")
+
 (let (globals (get-globals-sorted)
       globals-len (len globals)
-      i 0)
-    (load "init.slosh")
+      i 0
+      failing-tests #| {:dec! "dec!"
+                     :inc! "inc!"
+                     :xar! "xar!"
+                     :xdr! "xdr!"} |# )
```

Comment:
this is how you avoid specific ones. I'll remove this before the merge.

Reply:
For dec! and inc! try removing the negative test with "xxx" as the second argument.  That is a compile error now so the entire test will fail to compile.  Not sure how the errors are working out but that is likely the issue, those usage sections simply won't compile in slosh (which is good since we are compiling inc!/dec! we know that second arg better be an int or at least possibly be an int...).

For xar!/xdr! it is no longer valid to use them on nil/().  That was a decision, supporting that adds complexity for a case I don't think is common (and is easily handled with fn if needed).  Open to discussion on this but for now you should remove usage that is not valid anymore.

Reply:
I did, still not working. Here's a simple repro:

```
slosh> (def i 1)
1
Error getting prompt: [compile]: $sh: No such file or directory (os error 2)
slosh> (inc! i)
ERROR: [rt]: INC: Can only INC an integer type, got Special(Interned { id: 11 }).
prompt line: 42 ip: 0x00000004
DEBUG> :abort
```

Reply:
xar! and xdr! are working though, and i fixed seq? so it no longer recognizes nil.

Reply:
This is an a bug in inc/dec and globals.  I will push a patch once I sort it.